core/browser_buffer.py

```diff
@@ -90,3 +90,9 @@ def scroll_to_begin(self):
 
     def scroll_to_bottom(self):
         self.eval_js("window.scrollBy(0, document.body.scrollHeight)")
+
+    def set_url_for_bookmark(self):
```

Comment:
We should define interface function
```
def set_bookmark(self):
    pass
```

in emacs-application-framework/core/buffer.py

And implement interface ```set_bookmark``` in core/browser_buffer.py

Reply:
I'm not as fluent on the python side can you do it?

Reply:
Example, search ```scroll``` interface in core/buffer.py, you can define an empty interface in core/buffer.py and don't need implement it, then you can call ```scroll``` interface in Elisp side with eaf-call "execute_function".

And check app/pdf-viewer file, you can found ```scroll``` implement in pdf-viewer/buffer.py

In bookmark, you can define 
```
def set_bookmark(self):
    pass
```

in core/buffer.py 

Then implmenet ```set_bookarm``` in browser_buffer.py as your current patch content.